File: 二叉树/102.二叉树的层次遍历.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
package main

func levelOrder(root *TreeNode) [][]int {
	res := [][]int{}
	if root == nil {
		return res
	}
	que := make([]*TreeNode, 0)
	que = append(que, root)
	for len(que) > 0 {
		n := len(que)
		arr := []int{}
		for i := 0; i < n; i++ {
			tmp := que[i]
			if tmp.Left != nil {
				que = append(que, tmp.Left)
			}
			if tmp.Right != nil {
				que = append(que, tmp.Right)
			}
			arr = append(arr, tmp.Val)
		}
		que = que[n:len(que)]
		res = append(res, arr)
	}
	return res
}

File: 二叉树/102.二叉树的层次遍历.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        if(root==null){
            return res;
        }
        Queue<TreeNode> que = new LinkedList<>();
        que.add(root);
        while(!que.isEmpty()){
            int size = que.size();
            List<Integer> list = new ArrayList<>();
            while(size-->0){
                TreeNode tmp = que.poll();
                if(tmp.left!=null){
                    que.add(tmp.left);
                }
                if(tmp.right!=null){
                    que.add(tmp.right);
                }
                list.add(tmp.val);
            }
            res.add(list);
        }
        return res;
    }
}